src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -197,7 +197,7 @@ public virtual bool CanCreateInstance(Type targetType)
         /// <param name="bindingContext">The <see cref="ModelBindingContext"/>.</param>
         /// <remarks>
         /// <para>
-        /// This method should be called only when <see cref="MvcOptions.AllowValidatingTopLevelNodes" /> is
```

Comment:
Should we also remove the implementations that handle `AllowValidatingTopLevelNodes` and just mark it obsolete and say that it doesn't do anything?

Reply:
@rynowak why make the additional breaking change?

Users can disable validation of top-level nodes in a piecemeal fashion:
- replace the `IModelBinderProvider`s
- replace the `IModelMetadataProvider` with something that doesn't extend `ModelMetadataProvider`, and / or
- replace the `IObjectModelValidator` with something that doesn't extend `ObjectModelValidator`

But, ignoring `AllowValidatingTopLevelNodes` in the model binders would mean top-level validation could only be partially disabled.

Reply:
We don't intend to allow you to disable validation for parameters. If you don't like it, don't put validation attributes on parameters. We did a compat switch for this in case we broke functionality - which we did 😆 

I'm not suggesting we break lots of APIs. I'm suggesting that where the property/arg appears, we ignore it and document that it is ignored.